pkg/test/framework/components/apps/kube.go

```diff
@@ -664,28 +666,26 @@ func (a *kubeApp) Call(e AppEndpoint, opts AppCallOptions) ([]*echo.ParsedRespon
 		return nil, err
 	}
 
-	if len(resp) != 1 {
-		return nil, fmt.Errorf("unexpected number of responses: %d", len(resp))
-	}
-	if !resp[0].IsOK() {
-		return nil, fmt.Errorf("unexpected response status code: %s", resp[0].Code)
-	}
-	if resp[0].Host != dstServiceName {
-		return nil, fmt.Errorf("unexpected host: %s", resp[0].Host)
-	}
-	if resp[0].Port != strconv.Itoa(dst.networkEndpoint.ServicePort.Port) {
-		return nil, fmt.Errorf("unexpected port: %s", resp[0].Port)
-	}
-
 	return resp, nil
 }
 
-func (a *kubeApp) CallOrFail(e AppEndpoint, opts AppCallOptions, t testing.TB) []*echo.ParsedResponse {
+func (a *kubeApp) ValidatedCall(e AppEndpoint, opts AppCallOptions) ([]*echo.ParsedResponse, error) {
```

Comment:
I think it is ok to have this call (I'll let @nmittler to further comment on it), but I'd recommend adding a new type for the call result, and adding an "IsValid" method on it.

Reply:
@howardjohn I think we can still keep `CallOrFail`, right? Why did we want to remove it? It's irrelevant to whether or not we're introducing a new function to check the response, right?

Reply:
Yeah, let's just keep it `CallOrFail`.

Also, we can probably just drop the validation code altogether and just have the test do that.  The only exception is that the length check should probably still be done (in `Call`), but should be corrected to:

```golang
if len(r) != opts.Count {
    return nil, fmt.Errorf("unexpected number of responses. Expected %d, received %d", opts.Count, len(r))
}
```